### ISSUE TITLE
How do I hide nodes for a given criteria ?

### ISSUE THREAD

**Issue by justduy** [frontend-collective/react-sortable-tree#401]
A functionality that would resemble a filter function, such that nodes not matching a certain criteria would not render, but still be in the treeData.

**Comment by gragland** [comment]
I'm doing something like that and just have a master data object from which I generate treeData based on different conditions. It does complicate things quite a bit though because that means in my tree callbacks I need to take the path it gives me and convert it to the correct path to the node within my master data object.. but it works!

**Comment by justduy** [comment]
@gragland yeah I have opted for the same solution, but its a bit unnecessarily complex to do it. Would be better to not just render the nodes. I tried using the `addNodeUnderParent()`, but it doesn't work properly for some reason. I might just have to do the same thing as you then.

**Comment by justduy** [comment]
@gragland what does your `onChange()` callback do then ? Because the only information regarding moved nodes are within `oMoveNode()`. To show that tree has changed you would have to do in `onChange()`. In the `onMoveNode()` it would be fairly straightforward to do what you did. But it would make a laggy effect where the node would move and then jump back and then to its new location after `onMoveNode()`

**Comment by gragland** [comment]
@justduy I currently don't use `onChange()` at all and use `onMoveNode()` to handle moving of nodes. I'm not seeing any kind of laggy effect. From what I can tell there's no reason there should be a lag since it just calls `this.props.onMoveNode()` right after `this.props.onChange()`: https://github.com/frontend-collective/react-sortable-tree/blob/master/src/react-sortable-tree.js#L271
